src/common/utils/allowed-resource.ts

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright (c) 2021 OpenLens Authors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import { ClusterStore } from "../cluster-store";
+import type { KubeResource } from "../rbac";
+import { getHostedClusterId } from "./cluster-id-url-parsing";
+
+export function isAllowedResource(resource: KubeResource | KubeResource[]) {
+  const resources = [resource].flat();
+  const cluster = ClusterStore.getInstance().getById(getHostedClusterId());
+
+  if (!cluster?.allowedResources) {
+    return false;
+  }
+
+  if (resources.length === 0) {
+    return true;
+  }
+
+  const allowedResources = new Set(cluster.allowedResources);
+
+  return resources.every(resource => allowedResources.has(resource));
```

Comment:
Why is this better than:
```suggestion
  return resources.every(resource => cluster.allowedResources.includes(resource));
```

Reply:
Generally because I would figure it is less work (though I don't have the data to back it up).